### PR TITLE
docs: Add change log scripts

### PR DESCRIPTION
**Pull request by calldelegation**
# Summary

This PR aims to standardize release notes across Fuel Labs, inspired by the automated release notes system used by the TypeScript SDK team. In a conversation with @digorithm, he mentioned that he currently sifts through all the PRs to extract the breaking change notes from each PR included in the release.

The scripts have been rewritten in Rust. To use them, simply specify these items in the env example and execute cargo run in your terminal

1. GITHUB_TOKEN
2. GITHUB_REPOSITORY_OWNER
3. GITHUB_REPOSITORY_NAME 

https://github.com/user-attachments/assets/07e8e495-f897-4a7d-a656-ad7b8404e2e6

Each PR that should be picked up by the script at the end of each release should have titles like:
- chore
- fix
- feat

An exclamation mark following the title indicates a breaking change. For example, bugs can be classified under `fix`, and CI bumps under `chore`, but this can be open for discussion.

Screenshot 2024-08-22 at 3 47 21 PM

Within the breaking change section, please specify `before` and `after` code blocks, along with a brief note to describe the breaking changes within each PR if applicable. For example, in this PR: [https://github.com/FuelLabs/fuels-rs/pull/1464](https://github.com/FuelLabs/fuels-rs/pull/1464)

It might look like this:

**Before:**

Before Screenshot

**After:**

After Screenshot

Finally "in this release we..." should be a one liner to describe your PR if you want it to show up in the Summary notes at the new release.

# Checklist

- [ ] All **changes** are **covered** by **tests** (or not applicable)
- [ ] All **changes** are **documented** (or not applicable)
- [ ] I **reviewed** the **entire PR** myself (preferably on GitHub UI)
- [ ] I **described** all **Breaking Changes** (or confirmed there are none)

### CODE REVIEW

File: scripts/change-log/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[package]
+name = "change-log"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+regex = "1"
+dotenv = "0.15"
+tokio = { version = "1", features = ["full"] }
+octocrab = "0.39"
```
Comment (Br1ght0ne) [inline scripts/change-log/Cargo.toml:10]: Makes the script use exactly the same versions as all our other crates where possible, potentially saving compilation time:

```suggestion
[dependencies]
regex = { workspace = true }
dotenv = "0.15"
tokio = { workspace = true, features = ["full"] }
octocrab = "0.39"
```

File: .env.example [hunk 2]
```diff
@@ -0,0 +1,3 @@
+GITHUB_TOKEN
+GITHUB_REPOSITORY_OWNER
+GITHUB_REPOSITORY_NAME
```
Comment (Br1ght0ne) [inline .env.example:3]: Makes the format slightly clearer:

```suggestion
GITHUB_TOKEN=
GITHUB_REPOSITORY_OWNER=
GITHUB_REPOSITORY_NAME=
```

**Review by Br1ght0ne — COMMENTED** [review]
LGTM, thanks @calldelegation! Left a few nits, but the script itself looks great.

**Review by digorithm — APPROVED** [review]
Thanks for doing this, this will make our lives a lot easier when it comes to release notes. Excited to see this in action!

**Review by hal3e — APPROVED** [review]

**Pull request merged**